ambry-account/src/main/java/com/github/ambry/account/HelixAccountService.java

```diff
@@ -278,15 +293,31 @@ private void readFullAccountAndUpdateCache(String pathToFullAccountMetadata) thr
           logger.trace("Start parsing remote account data.");
           AccountInfoMap newAccountInfoMap = new AccountInfoMap(remoteAccountMap);
           Map<Short, Account> oldIdToAccountMap = accountInfoMapRef.get().idToAccountMap;
-          List<Short> idsOfUpdatedAccounts = new ArrayList<>();
+          accountInfoMapRef.set(newAccountInfoMap);
+          Map<Short, Account> idToUpdatedAccounts = new HashMap<>();
           for (Account newAccount : newAccountInfoMap.getAccounts()) {
             if (!newAccount.equals(oldIdToAccountMap.get(newAccount.getId()))) {
-              idsOfUpdatedAccounts.add(newAccount.getId());
+              idToUpdatedAccounts.put(newAccount.getId(), newAccount);
             }
           }
-          accountInfoMapRef.set(newAccountInfoMap);
-          logger.info("Received updates for {} accounts. Account IDs={}", idsOfUpdatedAccounts.size(),
-              idsOfUpdatedAccounts);
+          if (idToUpdatedAccounts.size() > 0) {
+            logger.info("Received updates for {} accounts. Account IDs={}", idToUpdatedAccounts.size(),
+                idToUpdatedAccounts.keySet());
+            for (Consumer<Collection<Account>> accountUpdateConsumer : accountUpdateConsumers) {
+              long startTime = System.currentTimeMillis();
+              try {
+                accountUpdateConsumer.accept(Collections.unmodifiableCollection(idToUpdatedAccounts.values()));
```

Comment:
minor: you can create one unmodifiable collection and pass it to everyone - creating a new object everytime is unnecessary.